OpenSSL_version.mk

```diff
@@ -1,80 +1,107 @@
+    $(info [TRACE] After OPENSSL_DIR set: LDFLAGS=$(LDFLAGS))
+    $(info [TRACE] After OPENSSL_DIR set: LDLIBS=$(LDLIBS))
 
-default: build show clean
+    .PHONY: default build show clean
 
-build: OpenSSL_version
+    default: build show clean
 
-show:	build
+    build: OpenSSL_version
+
+    show: build
 	@./OpenSSL_version$(EXE)
 
-clean:
+    clean:
 	@rm -f OpenSSL_version$(EXE)
 
-
 else ifeq ($(LIB),header)
+    $(info [DEBUG] LIB is set to header)
 
+    OPENSSL_NUMBER_SEL = head -n 1 | sed -r 's/.*OpenSSL //' | awk '{print ($$0+0)}'
+    OPENSSLV_H = $(OPENSSL_DIR)/include/openssl/opensslv.h
+    ifeq ($(shell fgrep OPENSSL_VERSION_MAJOR "$(OPENSSLV_H)"),)
+        $(info [DEBUG] OPENSSL_VERSION_MAJOR not found in OPENSSLV_H)
+        OPENSSL_VERSION = $(shell grep 'OPENSSL_VERSION_TEXT\s* "OpenSSL ' "$(OPENSSLV_H)" | $(OPENSSL_NUMBER_SEL))
+    else
+        $(info [DEBUG] OPENSSL_VERSION_MAJOR found in OPENSSLV_H)
+        ifeq ($(OS),MacOS)
+            $(info [DEBUG] OS is MacOS)
+            OPENSSL_VERSION = $(shell fgrep OPENSSL_VERSION_M "$(OPENSSLV_H)" | head -n 2 | awk -v RS="" '{print $4"."$8 }')
+        else
+            $(info [DEBUG] OS is not MacOS)
+            OPENSSL_VERSION = $(shell fgrep OPENSSL_VERSION_M "$(OPENSSLV_H)" | head -n 2 | awk -v RS="" '{print $$4"."$$8 }')
+        endif
+        $(info [TRACE] After OS check in header: OPENSSL_VERSION=$(OPENSSL_VERSION))
+    endif
 
-OPENSSL_NUMBER_SEL=head -n 1 | sed -r 's/.*OpenSSL //' | awk '{print ($$0+0)}'
-OPENSSLV_H=$(OPENSSL_DIR)/include/openssl/opensslv.h
-ifeq ($(shell fgrep OPENSSL_VERSION_MAJOR "$(OPENSSLV_H)"),)
-  OPENSSL_VERSION=$(shell grep 'OPENSSL_VERSION_TEXT\s* "OpenSSL ' "$(OPENSSLV_H)" | $(OPENSSL_NUMBER_SEL))
-else
-  ifeq ($(OS),MacOS)
-    OPENSSL_VERSION=$(shell fgrep OPENSSL_VERSION_M "$(OPENSSLV_H)" | head -n 2 | awk -v RS="" '{print $4"."$8 }')
-  else
-    OPENSSL_VERSION=$(shell fgrep OPENSSL_VERSION_M "$(OPENSSLV_H)" | head -n 2 | awk -v RS="" '{print $$4"."$$8 }')
-  endif
-endif
-
-ifeq ($(OPENSSL_VERSION),1)
-    OPENSSL_VERSION=1.0
-endif
+    ifeq ($(OPENSSL_VERSION),1)
+        $(info [DEBUG] OPENSSL_VERSION is 1)
+        OPENSSL_VERSION = 1.0
+    endif
 
+    $(info [TRACE] OPENSSL_NUMBER_SEL=$(OPENSSL_NUMBER_SEL))
+    $(info [TRACE] OPENSSLV_H=$(OPENSSLV_H))
+    $(info [TRACE] OPENSSL_VERSION=$(OPENSSL_VERSION))
 
 else # $(LIB) is name of library file
+    $(info [DEBUG] LIB is set to a library file)
 
+    OPENSSL_VERSION = $(shell strings "$(LIB)" | grep -E 'OpenSSL [0-9]+\.[0-9]+\.' | head -n 1 | sed -r 's/.*OpenSSL //' | awk -v FS="." '{print $$1"."$$2}')
+    ifeq ($(OPENSSL_VERSION),)
+        $(info [DEBUG] OPENSSL_VERSION not found in library file)
+        OPENSSL_VERSION = $(shell strings "$(LIB)" | grep -E 'libcrypto\.' | head -n 1 | sed -r 's/.*libcrypto(.[a-z]+)?\.//')
+    endif
+    ifeq ($(OPENSSL_VERSION),1.0.0)
+        $(info [DEBUG] OPENSSL_VERSION is 1.0.0)
```

Comment:
this line is not needed due to version output below (line 100)

Reply:
fixed